src/megaclient.cpp

```diff
@@ -13268,8 +13315,9 @@ bool MegaClient::syncdown(LocalNode* l, LocalPath& localpath, bool rubbish)
                 }
 
                 // recurse into directories of equal name
-                if (!syncdown(ll, localpath, rubbish) && success)
+                if (!syncdown(ll, localpath, scanWholeSubtree))
                 {
+                    l->syncdownTargetedAction = std::max<int>(originalTargetedAction, LocalNode::synctree_descendantflagged);
```

Comment:
If both values are unsigned, I think we should use `max<unsigned>`

Reply:
updated

Reply:
I will change the other one too, which is equivalent to this one: https://github.com/meganz/sdk/pull/2198/files#diff-29ce12163767d4d38759d549fa619b81R13510